src/read.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2021 by Accenture ESR
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # dlt reading support
+use crate::{
+    dlt::{HEADER_MIN_LENGTH, STORAGE_HEADER_LENGTH},
+    filtering::ProcessedDltFilterConfig,
+    parse::{dlt_message, parse_length, DltParseError, ParsedMessage},
+};
+use std::io::{BufReader, Read};
+
+const DEFAULT_BUFFER_CAPACITY: usize = 10 * 1024 * 1024;
+const DEFAULT_MESSAGE_MAX_LEN: usize = 10 * 1024;
```

Comment:
is 10k enough for normal usecases?

Reply:
Changed the DEFAULT_MESSAGE_MAX_LEN to the maximum possible length of a DLT message (u16::MAX) + length of a storage header.